runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDoTranslation.java

```diff
@@ -496,6 +497,25 @@ private static SdkFunctionSpec toProto(ViewFn<?, ?> viewFn) {
         .build();
   }
 
+  private static <T> ParDoPayload getParDoPayload(AppliedPTransform<?, ?, ?> transform)
+      throws IOException {
+    return PTransformTranslation.toProto(
```

Comment:
It is definitely circuitous. I wanted to support direct `Pipeline`-to-runner interactions as well as rehydrated pipelines. There are a couple other options:

1. Don't support that. Make every runner operate only on rehydrated `Pipeline` filled only with `RawPTrasnform`. That is probably a win for portability, but somewhat difficult in transition.
2. Get rid of all the static methods for pulling stuff out, and just have a `toRawPTransform` method; let runners shuffle proto fields. Eventually they may need to anyhow.
3. Add wrapper `RawPTransform` subclasses for each of the `XYZTranslation` classes with all the same methods that are currently static methods. These could be the result of dehydration or wrap a native Java class, so you still get the direct-to-runner capability.